tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -75,25 +84,47 @@ def finalizer():
                 taint_label="xyz=true:NoSchedule",
             ), "Failed to untaint"
 
+            resource_name = constants.DEFAULT_CLUSTERNAME
+            if config.DEPLOYMENT["external_mode"]:
+                resource_name = constants.DEFAULT_CLUSTERNAME_EXTERNAL_MODE
+
+            logger.info("Remove tolerations from storagecluster")
+            storagecluster_obj = ocp.OCP(
+                resource_name=resource_name,
+                namespace=config.ENV_DATA["cluster_namespace"],
+                kind=constants.STORAGECLUSTER,
+            )
+            params = '[{"op": "remove", "path": "/spec/placement"},]'
+            storagecluster_obj.patch(params=params, format_type="json")
+
+            logger.info("Remove tolerations from subscriptions")
+            sub_list = ocp.get_all_resource_names_of_a_kind(kind=constants.SUBSCRIPTION)
+
+            sub_obj = ocp.OCP(
+                namespace=config.ENV_DATA["cluster_namespace"],
+                kind=constants.SUBSCRIPTION,
+            )
+            for sub in sub_list:
+                subscription_data = sub_obj.get(resource_name=sub)
+                if "config" in subscription_data.get("spec", {}):
+                    params = '[{"op": "remove", "path": "/spec/config"}]'
+                    sub_obj.patch(resource_name=sub, params=params, format_type="json")
+            time.sleep(180)
+            assert wait_for_pods_to_be_running(
+                timeout=900, sleep=15
+            ), "Few pods failed to reach the desired running state"
+
         request.addfinalizer(finalizer)
 
-    def test_non_ocs_taint_and_tolerations(self):
+    def apply_custom_taint_and_toleration(self):
```

Comment:
should we move it to some common place outside test? 
You can have an arg with taint_label to this function